File: M3Tools/Dialogs/Emails/ReciepientSelection.vb
﻿Imports System.Windows.Forms

Public Class ReciepientSelection

	ReadOnly Property Selection As Types.DBEntryCollection(Of Types.Listener)
		Get
			Dim list As New Types.DBEntryCollection(Of Types.Listener)()

			For Each listener As DataGridViewRow In ldg_Listeners.SelectedListeners
				list.Add(CType(listener.DataBoundItem, Types.Listener))
			Next

			Return list
		End Get
	End Property

	Public Property DataSource As BindingSource
		Get
			Return ldg_Listeners.DataSource
		End Get
		Set(value As BindingSource)
			ldg_Listeners.DataSource = value
		End Set
	End Property


	Private Sub ConfirmSelection(sender As Object, e As EventArgs) Handles btn_Select.Click
		If Selection.Count < 1 Then
			MessageBox.Show("You didn't select a listener. If you wish to cancel, please click the cancel button", "No Selection", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Return
		End If

		Me.DialogResult = DialogResult.OK
		Me.Close()
	End Sub

	Private Sub Cancel(sender As Object, e As EventArgs) Handles btn_Cancel.Click
		Me.DialogResult = DialogResult.Cancel
		Me.Close()
	End Sub
End Class

File: M3Tools/Dialogs/Emails/ReciepientSelection.cs
using System;
using System.Windows.Forms;

namespace SPPBC.M3Tools
{

    public partial class ReciepientSelection
    {

		/// <summary>
		/// The current selection of recipients
		/// </summary>
        public Types.DBEntryCollection<Types.Listener> Selection
        {
            get
            {
                return (Types.DBEntryCollection<Types.Listener>)ldg_Listeners.SelectedRows;
            }
        }

		/// <summary>
		/// The data source for the control
		/// </summary>
        public Data.ListenerBindingSource DataSource
        {
            get
            {
                return ldg_Listeners.DataSource as Data.ListenerBindingSource;
            }
            set
            {
                ldg_Listeners.DataSource = value;
            }
        }

		/// <summary>
		/// 
		/// </summary>
        public ReciepientSelection()
        {
            InitializeComponent();
        }


        private void ConfirmSelection(object sender, EventArgs e)
        {
            if (Selection.Count < 1)
            {
                MessageBox.Show("You didn't select a listener. If you wish to cancel, please click the cancel button", "No Selection", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return;
            }

            DialogResult = DialogResult.OK;
            Close();
        }

        private void Cancel(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }
    }
}
